### PR TITLE
[util] Prevent iteration waiting in duration waiter if condition reached

### PR DESCRIPTION
**Pull request by draker94**

### CODE REVIEW

**Review by valfirst — APPROVED** [review]

File: vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java [hunk 1]
```diff
@@ -50,14 +50,18 @@ public <T, E extends Exception> T wait(FailableSupplier<T, E> valueProvider, Pre
             long iterationStartTime = System.currentTimeMillis();
             value = valueProvider.get();
             long iterationEndTime = System.currentTimeMillis();
+            if (stopCondition.test(value))
+            {
+                break;
+            }
```
Comment (valfirst) [inline vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java:56]:
```suggestion
            if (stopCondition.test(value))
            {
                break;
            }
            long iterationEndTime = System.currentTimeMillis();
```

Comment (draker94) [reply vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java]: This is how it was at the beginning
And we again get **this** error

image

Comment (valfirst) [reply vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java]: I guess this test is needed to be tuned

Comment (valfirst) [reply vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java]: this is expected, since we're changing the underneath logic

**Review by uarlouski — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**